product_pricelist_alternative/models/product_pricelist.py

```diff
@@ -67,6 +67,9 @@ def _compute_price_rule(
         compute_price=True,
         **kwargs,
     ):
+        if self.env.context.get("force_price_recomputation"):
+            compute_price = True
```

Comment:
IMO you don't need to depend on `sale` for adding such code based on a context key.